cgen error when using `import v.reflection` and a struct with field attributes, that have double quotes

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

**V doctor:**

```
V full version: V 0.4.8 aeaf607.f2c281e
OS: linux, Ubuntu 20.04.6 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /space/v/oo
vexe: /home/delian/code/v/v
vexe mtime: 2024-11-29 08:15:02

vroot: OK, value: /home/delian/code/v
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.46.0
Git vroot status: weekly.2024.48-24-g7a3e7170-dirty (16 commit(s) behind V master)
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
emcc version: N/A
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9
```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg v_reflection_on_attributes_with_double_quotes_cgen_bug.v`

```v
import v.reflection as _

struct MyParams {
     p_fpga_ver  string  @[long: fp_ver; name: 'FPGA Version'; xdoc:'String to use as simulated FPGA version in Version responses. Must be in the form "a.bb.cccc"']
     p_cm_ver    string  @[long: cm_ver; name: 'CM Version'; xdoc:'String to use as simulated CM version in Version responses. Must be in the form "a.bb.cccc"']
}     
```

**What did you expect to see?**

A compiled program

**What did you see instead?**

```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/v_reflection_on_attributes_with_double_quotes_cgen_bug.01JE335BE1XPJNKQN1ZZK4TWE5.tmp.c:14625: error: ')' expected (got "a")
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzRkNWQ0MWQyZWY0Y2JjYzQ2ZDUxYmQiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.vb7IlPRf8UbzgRI2xWiCTBvxO1hh_OfcDPbd2EaJJgg">Huly&reg;: <b>V_0.6-21486</b></a></sub>

--- a/vlib/v/gen/c/reflection.v
+++ b/vlib/v/gen/c/reflection.v
@@ -95,7 +95,11 @@ fn (g &Gen) gen_attrs_array(attrs []ast.Attr) string {
 	}
 	mut out := 'new_array_from_c_array(${attrs.len},${attrs.len},sizeof(string),'
 	out += '_MOV((string[${attrs.len}]){'
-	out += attrs.map(if it.has_arg { '_SLIT("${it.name}=${it.arg}")' } else { '_SLIT("${it.name}")' }).join(',')
+	out += attrs.map(if it.has_arg {
+		'_SLIT("${it.name}=${escape_quotes(it.arg)}")'
+	} else {
+		'_SLIT("${it.name}")'
+	}).join(',')
 	out += '}))'
 	return out
 }

--- a/vlib/v/tests/reflection_attr_quotes_test.v
+++ b/vlib/v/tests/reflection_attr_quotes_test.v
@@ -1,0 +1,16 @@
+import v.reflection as r
+
+struct MyParams {
+	p_fpga_ver string @[long: fp_ver; name: 'FPGA Version'; xdoc: 'String to use as simulated FPGA version in Version responses. Must be in the form "a.bb.cccc"']
+	p_cm_ver   string @[long: cm_ver; name: 'CM Version'; xdoc: 'String to use as simulated CM version in Version responses. Must be in the form "a.bb.cccc"']
+}
+
+fn test_main() {
+	a := MyParams{}
+	t := r.type_of(a)
+	if t.sym.info is r.Struct {
+		assert t.sym.info.fields[0].attrs[2] == 'xdoc=String to use as simulated FPGA version in Version responses. Must be in the form "a.bb.cccc"'
+	} else {
+		assert false
+	}
+}

